internal/mkcw/archive.go

```diff
@@ -245,6 +244,38 @@ func Archive(path string, ociConfig *v1.Image, options ArchiveOptions) (io.ReadC
 	// Pretend that we didn't have to do any of the preceding.
 	path = rootfsDir
 
+	// Write extra content to the rootfs, creating intermediate directories if necessary.
+	for location, content := range options.ExtraImageContent {
+		err := func() error {
+			if err := idtools.MkdirAllAndChownNew(filepath.Dir(filepath.Join(path, location)), 0o755, idtools.IDPair{UID: int(st.UID()), GID: int(st.GID())}); err != nil {
+				return fmt.Errorf("ensuring %q is present in container root filesystem: %w", filepath.Dir(location), err)
+			}
+			output, err := os.OpenFile(filepath.Join(path, location), os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
+			if err != nil {
+				return fmt.Errorf("preparing to write %q to container root filesystem: %w", location, err)
+			}
+			defer output.Close()
+			input, err := os.Open(content)
+			if err != nil {
+				return err
+			}
+			defer input.Close()
+			if _, err := io.Copy(output, input); err != nil {
+				return fmt.Errorf("copying contents of %q to %q in root filesystem: %w", location, content, err)
```

Comment:
Above error stmt is using `container root filesystem` instead of only `root filesystem`, maybe we can just keep any one of them for consistency.

```suggestion
				return fmt.Errorf("copying contents of %q to %q in container root filesystem: %w", location, content, err)
```

Reply:
Changing it.  Also, I had the orders of `location` and `content` mixed up, switching them.